build_system/seqan3-config.cmake

```diff
@@ -359,6 +359,7 @@ else ()
             set (C++17FS_LIB ${_LIB})
             break ()
         endif ()
+        set (CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES_ORIGINAL})
     endforeach ()
```

Comment:
We could do:

    foreach (_LIB stdc++fs)
        set (CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES_ORIGINAL} ${_LIB})

        check_cxx_source_compiles ("${CXXSTD_TEST_SOURCE}" C++17FS_LIB-l${_LIB})

        if (C++17FS_LIB-l${_LIB})
            set (SEQAN3_LIBRARIES ${SEQAN3_LIBRARIES} ${_LIB})
            set (C++17FS_LIB ${_LIB})
            break ()
        endif ()
    endforeach ()

    if (NOT C++17FS_LIB)
        if(Boost_FOUND)
         set (SEQAN3_LIBRARIES    ${SEQAN3_LIBRARIES}    ${Boost_LIBRARIES}) # SYSTEM shuld avoid warning, but not
         set (SEQAN3_INCLUDE_DIRS ${SEQAN3_INCLUDE_DIRS} ${Boost_INCLUDE_DIR})
         set (CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES_ORIGINAL} ${Boost_LIBRARIES}) #ORIGINAL!!
         set (CMAKE_REQUIRED_INCLUDES  ${CMAKE_REQUIRED_INCLUDES}  ${Boost_LIBRARIES})
         set (SEQAN3_INCLUDE_DIRS ${SEQAN3_INCLUDE_DIRS} ${Boost_INCLUDE_DIR})
         set (C++17FS_LIB         ${Boost_LIBRARIES})
        endif ()
    endif()

but I think it could do it harder to understand in the future:

Reply:
But we don't include any boost?